### PR TITLE
Refactoring CSS: Using Pseudo-elements selector

### PR DESCRIPTION
**Pull request by stevenjoezhang**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [ ] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] [Docs](https://github.com/theme-next/theme-next.org/tree/source/source/docs) in [NexT website](https://theme-next.org/docs/) have been added / updated (for features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [x] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue resolved: N/A

## What is the new behavior?

- Screenshots with this changes: N/A
- Link to demo site with this changes: N/A

### How to use?
In NexT `_config.yml`:
```yml
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

File: source/css/_common/components/post/post.styl [hunk 1]
```diff
@@ -46,7 +46,6 @@
 
 @import "post-expand";
 @import "post-collapse";
-@import "post-type";
```
Comment (1v9) [inline source/css/_common/components/post/post.styl]: unneeded?

File: source/css/_common/components/pages/categories.styl [hunk 2]
```diff
@@ -2,26 +2,4 @@
   .category-all-title { text-align: center; }
 
   .category-all { margin-top: 20px; }
-
```
Comment (1v9) [inline source/css/_common/components/pages/categories.styl]: Why remove...

Comment (stevenjoezhang) [reply source/css/_common/components/pages/categories.styl]: The selector matches nothing...
You can confirm that

File: source/css/_common/components/post/post.styl (hunk 1 above)
Comment (stevenjoezhang) [reply source/css/_common/components/post/post.styl]: `.post-type-quote` in `post-type.styl` no longer exist, so removed it

File: source/css/_common/components/pages/categories.styl (hunk 2 above)
Comment (1v9) [reply source/css/_common/components/pages/categories.styl]: Are you sure?

**Comment by stevenjoezhang** [comment]
Oh, how careless I am ... This class name is spliced out
屏幕快照 2019-07-09 下午4 10 30

File: source/css/_common/components/post/post-type.styl [hunk 3]
```diff
@@ -1,14 +0,0 @@
-// TODO: Refactor.
-
-.page-home, .page-post-detail {
-  .post-type-quote {
```
Comment (stevenjoezhang) [inline source/css/_common/components/post/post-type.styl]: @1v9 you can help confirm if `.post-type-quote` still exist

Comment (1v9) [reply source/css/_common/components/post/post-type.styl]: @stevenjoezhang OK

File: source/css/_common/components/post/post.styl (hunk 1 above)
Comment (1v9) [reply source/css/_common/components/post/post.styl]: https://github.com/theme-next/hexo-theme-next/blob/243804ad75ec4ff146ed7225c46264e72cbe8e8b/layout/_macro/post.swig#L37-L38

**Comment by stevenjoezhang** [comment]
I see. Now we got several page/post types:
post: picture | quote
page: categories | tags | schedule

**Comment by 1v9** [comment]
@stevenjoezhang Yes, and `// TODO: Refactor.` is right 😅

File: source/css/_common/components/tags/tabs.styl [hunk 4]
```diff
@@ -22,6 +22,7 @@
     li.tab {
       flex-grow: 1;
       list-style-type: none;
+      background-color: #fff;
```
Comment (1v9) [inline source/css/_common/components/tags/tabs.styl]: This is the last question, what's this for?

Comment (stevenjoezhang) [reply source/css/_common/components/tags/tabs.styl]: It was deleted in #704. Then I found that in some browser, `border-bottom` won't work correctly without a `background-color` parameter (e.g. `border-bottom: transparent` or `border-bottom: initial`), so add it back

Comment (1v9) [reply source/css/_common/components/tags/tabs.styl]: 反复横跳😅

**Comment by stevenjoezhang** [comment]
I'm still confused by `post-type-quote` 😂
Let's deal with it later

**Comment by 1v9** [comment]
@stevenjoezhang Low Low priority ✔

**Pull request merged**